### ISSUE TITLE
Is it possible to make the same sails instance accessible with http and https?

### ISSUE THREAD

**Issue by JumpLink** [balderdashy/sails#3118]
I want to prefer https but as a fallback I want to still support http but want to run only one instance of sails. Is this possible?

**Comment by Dallas62** [comment]
Hi!
Use only one instance of Sails over http and put a reverse proxy (like Nginx or HAProxy) in front of your instance.
you will be able to have http and https on the "same" instance, and the reverse proxy will handle both http and https requests.

Internet <===> Nginx (port: 80 / 443) <===> sails.js (port: 8080 or whatever).

After your reverse proxy, the request will not be encrypt.

**Comment by tjwebb** [comment]
> Use only one instance of Sails over http and put a reverse proxy (like Nginx or HAProxy) in front of your instance.

yes. Never expose a node.js server directly to the internet.

**Comment by sailsbot** [comment]
Thanks for posting, @JumpLink.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
